internal/pkg/term/progress/render.go

```diff
@@ -14,6 +22,28 @@ type Renderer interface {
 	Render(out io.Writer) (numLines int, err error)
 }
 
+// Render renders r periodically to out until the ctx is canceled or an error occurs.
+// While Render is executing, the terminal cursor is hidden and moved up after each call so that
+// the updates are written in-place.
+func Render(ctx context.Context, out io.Writer, r Renderer) error {
+	cursor := cursor.NewWithWriter(out)
+	cursor.Hide()
+	defer cursor.Show()
+	for {
+		select {
+		case <-ctx.Done():
+			_, err := r.Render(out)
+			return err
+		case <-time.After(renderInterval):
+			nl, err := r.Render(out)
+			if err != nil {
+				return err
+			}
+			cursor.Up(nl + 1) // move the cursor back up to the starting line so that the Renderer is rendered in-place.
```

Comment:
Does this work as expected for Linux and Windows?

Reply:
I haven't checked it on windows yet! I'm thinking of getting the display working e2e on Mac and Linux first and then readjusting according on windows